api/edge_api/identities/views.py

```diff
@@ -300,6 +306,29 @@ def all(self, request, *args, **kwargs):
 
         return Response(serializer.data)
 
+    @swagger_auto_schema(request_body=EdgeIdentitySourceIdentityRequestSerializer())
+    @action(detail=False, methods=["POST"], url_path="clone-from-given-identity")
+    def clone_from_given_identity(self, request, *args, **kwargs) -> Response:
+        """
+        Clone feature states from a given source identity.
+        """
+        # Get and validate source identity
+        serializer = EdgeIdentitySourceIdentityRequestSerializer(
+            data=request.data, context={"request": request}
+        )
+        serializer.is_valid(raise_exception=True)
+
+        source_identity: EdgeIdentity = self.get_identity(
+            edge_identity_identity_uuid=request.data.get("source_identity_uuid")
+        )
+
+        self.identity.clone_flag_states_from(source_identity)
+        self.identity.save(
+            user=request.user.id,
+            master_api_key=getattr(request, "master_api_key", None),
+        )
+        return Response(status=status.HTTP_200_OK)
```

Comment:
Yeah, I was originally sending the new Identity feature states but in the FE that would require a lot of additional logic, currently, the API client we use (RTK Query) invalidates the state to build the view, and then calls the endpoint originally used to get the data view, the get all feature states, and while we could tweak it,  that's the RTK's philosophy.

Reply:
It seems strange to me that RTK can't use data included in the response.

Reply:
It can, I just would need extra logic to merge the data coming from the original endpoint which includes much more data and this one, while in the RTKway you just invalidate the data being changed and it will make the request, update the view, or pick it from cache, or do whatever is needed automatically.

Reply:
Ok, I'm not sure that necessarily means that the view shouldn't return the data though - other direct API users might want to see that data.

Reply:
Right. Returning the identity feature flags again.